Google.Api.Gax.Grpc.Gcp/DefaultChannelCredentialsCache.cs

```diff
@@ -54,12 +59,22 @@ internal DefaultChannelCredentialsCache(IEnumerable<string> scopes)
                     {
                         appDefaultCredentials = appDefaultCredentials.CreateScoped(_scopes);
                     }
+
+                    if (_useUseJwtAccessWithScopes && appDefaultCredentials.UnderlyingCredential is ServiceAccountCredential)
+                    {
+                        ServiceAccountCredential serviceCredential = appDefaultCredentials.UnderlyingCredential as ServiceAccountCredential;
+                        appDefaultCredentials = GoogleCredential.FromServiceAccountCredential(serviceCredential.WithUseJwtAccessWithScopes(true));
+                    }
                     return appDefaultCredentials.ToChannelCredentials();
                 }));
         }
 
+        internal DefaultChannelCredentialsCache(IEnumerable<string> scopes) : this(scopes, false)
```

Comment:
Do we need this internal overload, or can we just call the other one always specifiying true or false?

Reply:
Also, this whole class is internal. But you haven't changed code using this class, and by passing false here, there's effectively no change in behaviour at all. Is this in preparation for when jwts are the default? Then I'd rather we move that preparation to client code. Keep here just one constructor and have the client class pass true or false depending on what's the default for it.

Reply:
I was not sure about this one yet. Now I see that it is used only by GcpCallInvokerPool and that class isn't referenced by Gapic generator. It probably mean this class is out of scope for now? 

Or if we want to provide general functionality to enable SS Jwt if needed, it means, as you suggested, I can modify GcpCallInvokerPool accordingly and explicitly pass true/false into here.

Reply:
GcpCallInvokerPool is only used by Spanner and PubSub, I believe. (It should probably go away entirely in the next version of GAX, to be honest.)

I need to have a think about how all of this works with channel pools in general... I'll *try* to do that this afternoon, but unfortunately it may need to wait until late next week. (I'm on vacation from tomorrow until Wednesday afternoon.)

Reply:
I have updated changes to this file, as its hard for me to estimate how soon it will go away. The solution is basically the same, so not much overhead to simply have it. Let me know if it is safe to assume this code as gone and skip it :)